lib/messageformat.js

```diff
@@ -363,5 +363,5 @@ MessageFormat.prototype.compile = function(messages, locale) {
   return result;
 }
 
-
+MessageFormat.MessageFormat = MessageFormat;
```

Comment:
This looks pretty weird to me. Is this really a common pattern for TypeScript compatibility with CommonJS exports? Would a TS user really expect for `import { MessageFormat } from 'messageformat'` to "just work", or would it need to be documented?

Reply:
This is more about es6 modules than Typescript. With es6 imports, I expect to import a module containing classes and functions and not a class or a function directly (like the fs module in node).

In Angular (typescript) every import looks like `import { MyClass } from './myClass.ts'`

When doing this consequently with require one would have to do
```javascript
const MessageFormat = require('messageformat').MessageFormat;
```
as equivalent to
```typescript
import { MessageFormat } from 'messageformat';
```

To get this working and also keep it working with the simple require, I did this "pretty weird" hack... And because it also looked not so beautiful to me, I did this as a second commit, that can be rejected ;)

For my needs it would be enough to have the first commit, that enables me to do

```typescript
import * as MessageFormat from 'messageformat';
```
because I can not use require in my angular project